navigation/navigation-runtime/build.gradle

```diff
@@ -123,33 +108,52 @@ kotlin {
         }
 
         jbMain.dependsOn(commonMain)
+        jbTest.dependsOn(commonTest)
 
         nonJvmMain {
             dependsOn(jbMain)
             dependencies {
                 implementation(libs.atomicFu)
             }
         }
+        nonJvmTest.dependsOn(jbTest)
 
         desktopMain {
             dependsOn(jvmMain)
             dependsOn(jbMain)
         }
+        desktopTest {
+            dependsOn(jvmTest)
+            dependsOn(jbTest)
+        }
+
+        webMain.dependsOn(nonJvmMain)
+        webTest.dependsOn(nonJvmTest)
 
         targets.all { target ->
-            if (target.platformType !in [
-                    KotlinPlatformType.androidJvm,
-                    KotlinPlatformType.jvm,
-                    KotlinPlatformType.common
-            ]) {
-                target.compilations["main"].defaultSourceSet {
-                    dependsOn(nonJvmMain)
-                }
+            if (target.platformType == KotlinPlatformType.native) {
+                target.compilations["main"].defaultSourceSet.dependsOn(nonJvmMain)
+                target.compilations["test"].defaultSourceSet.dependsOn(nonJvmTest)
+            } else if (target.platformType in [KotlinPlatformType.js, KotlinPlatformType.wasm]) {
+                target.compilations["main"].defaultSourceSet.dependsOn(webMain)
+                target.compilations["test"].defaultSourceSet.dependsOn(webTest)
             }
         }
+
+        wasmJsTest.kotlin.srcDirs("src/jsTest/kotlin")
     }
 }
 
+kotlin {
+    watchosArm64()
```

Comment:
I see we already had those targets added for navigation-common and navigation-runtime.
I just wanted to say that publishing all these extra targets requires more time for our CI to make a full build. 


At the same time, compose-ui, compose-foundation and others do not have these targets.
I understand that navigation library is independent from Compose UI. But do we see any use cases when it's used without Compose UI?

If not, do we want to consider skipping these targets until we know there is a demand for them?

cc: @MatkovIvan

Reply:
Agree in general, but I'd suggest making it as a separate, dedicated to only that PR